lib/ansible/plugins/inventory/kubevirt.py

```diff
@@ -0,0 +1,238 @@
+except ImportError:
+    pass
+
+
+API_VERSION = 'kubevirt.io/v1alpha3'
+
+
+class InventoryModule(K8sInventoryModule):
+    NAME = 'kubevirt'
+
+    def fetch_objects(self, connections):
+        client = self.get_api_client()
+
+        if connections:
+            for connection in connections:
+                client = self.get_api_client(**connection)
+                name = connection.get('name', self.get_default_host_name(client.configuration.host))
+                if connection.get('namespaces'):
+                    namespaces = connection['namespaces']
+                else:
+                    namespaces = self.get_available_namespaces(client)
+                interface_name = connection.get('network_name')
+                api_version = connection.get('api_version', API_VERSION)
+                for namespace in namespaces:
+                    self.get_vms_for_namespace(client, name, namespace, interface_name, api_version)
+        else:
+            name = self.get_default_host_name(client.configuration.host)
+            namespaces = self.get_available_namespaces(client)
+            for namespace in namespaces:
+                self.get_vms_for_namespace(client, name, namespace, None, api_version)
+
+    def get_vms_for_namespace(self, client, name, namespace, interface_name=None, api_version=None):
+        v1_vm = client.resources.get(api_version=api_version, kind='VirtualMachineInstance')
+        try:
+            obj = v1_vm.get(namespace=namespace)
+        except DynamicApiError as exc:
+            self.display.debug(exc)
+            raise K8sInventoryException('Error fetching Virtual Machines list: %s' % format_dynamic_api_exc(exc))
+
+        namespace_group = 'namespace_{0}'.format(namespace)
+        namespace_vms_group = '{0}_vms'.format(namespace_group)
+
+        name = self._sanitize_group_name(name)
+        namespace_group = self._sanitize_group_name(namespace_group)
+        namespace_vms_group = self._sanitize_group_name(namespace_vms_group)
+        self.inventory.add_group(name)
+        self.inventory.add_group(namespace_group)
+        self.inventory.add_child(name, namespace_group)
+        self.inventory.add_group(namespace_vms_group)
+        self.inventory.add_child(namespace_group, namespace_vms_group)
+        for vm in obj.items:
+            if not (vm.status and vm.status.interfaces):
+                continue
+
+            # Find interface by its name:
+            if interface_name is None:
+                interface = vm.status.interfaces[0]
+            else:
+                interface = next(
+                    (i for i in vm.status.interfaces if i.name == interface_name),
+                    None
+                )
+
+            # If interface is not found or IP address is not reported skip this VM:
+            if interface is None or interface.ipAddress is None:
+                continue
+
+            name_format = '{namespace}-{name}-{uid}'
```

Comment:
Could you move this string in the parameter list of the module?

Reply:
Done

Reply:
Thank you @machacekondra!